csharp/src/Microsoft.ML.OnnxRuntime/OrtIoBinding.cs

```diff
@@ -61,6 +73,75 @@ public void BindInput(string name, Tensors.TensorElementType elementType, long[]
+        /// ArrayUtilities.GetSizeForShape(shape) * sizeof(T) == bytesSize
+        /// Note, that the resulting OrtValue does not own the memory being passed. It is
+        /// a responsibility of the user to dispose of MemoryHandle instance after the binding is no longer
+        /// needed.
+        /// </summary>
+        /// <param name="name">input name</param>
+        /// <param name="elementType">TensorElementType</param>
+        /// <param name="shape">shape of the tensor to be created</param>
+        /// <param name="memoryInfo">memoryInfo. For managed buffers simply use OrtMemoryInfo.DefaultInstance
+        /// For non-cpu memory you will need to obtain an appropriate instance. See OrtMemoryInfo.
+        /// This is used to decide if the buffer needs to be copied to a device at the time of the binding.</param>
+        /// <param name="memHandle">handle obtained from Memory<T>.Pin()></param>
+        /// <param name="bytesSize">size of the allocation in bytes</param>
+        /// 
+        /// <example>
+        /// Here is an example of using a 3rd party library class for processing float16/bfloat16.
+        /// Currently, to pass tensor data and create a tensor one must copy data to Float16/BFloat16 structures
+        /// so DenseTensor can recognize it.
+        /// If you are using a library that has a class Half and it is blittable, that is its managed in memory representation
+        /// matches native one and its size is 16-bits, you can use the following conceptual example
+        /// to feed/fetch data for inference using Half array.
+        /// 
+        /// \code{.cs}
+        /// unsafe { Debug.Assert(sizeof(ushort) == sizeof(Half)); }
+        /// var ioBinding = session.CreateIoBinding(); // Needs disposal
+        /// Half[] input = new Half[] { 5646, 12345 };
+        /// Half[] output = new Half[40]; // Whatever the expected len/shape is must match
+        /// var input_shape = new long[] {input.Length};
+        /// var output_shape = new long[] { output.Length};
+        /// 
+        /// var memInfo = OrtMemoryInfo.DefaultInstance; // CPU
+        /// using(var inputHandle = new Memory<Half>(input).Pin())
+        /// using(var outputHandle = new Memory<Half>(output).Pin())
+        /// {
+        ///     ioBinding.BindInput("input_name", Tensors.TensorElementType.Float16, 
+        ///                          memInfo, input_handle, input.Length * sizeof(ushort));
+        ///     ioBinding.BindOutput("output_name", Tensors.TensorElementType.Float16, 
+        ///                          memInfo, output_handle, output.Length * sizeof(ushort));
+        ///     session.RunWithBinding(runOptions, ioBinding);
+        /// }
+        /// // Access your output using your preferred type Half
+        /// // Hint, you can use this for other supported types as well.
+        /// \endcode
+        /// </example>
+
+        public void BindInput(string name, Tensors.TensorElementType elementType, long[] shape, 
```

Comment:
So just to understand better -  the user would use this overload of BindInput() ahead of the overload of BindInput() that takes in a OrtMemoryAllocation when they have blittable types because otherwise they may have to do a native allocation and populate its contents (i.e.) introduce a "copy" ?

Reply:
Why can't we allow creating FixedBufferOnnxValue from a MemoryHandle and the other overload of BindInput can be used presumably